nimbus/src/enrollment.rs

```diff
@@ -85,90 +365,156 @@ pub fn get_enrollments(db: &Database) -> Result<Vec<EnrolledExperiment>> {
     Ok(result)
 }
 
-/// Update all enrollments. Typically used immediately after the list of
-/// experiments has been refreshed, so some might mean new enrollments, some
-/// might have expired, etc.
-pub fn update_enrollments(
-    db: &Database,
-    mut writer: &mut Writer,
-    nimbus_id: &Uuid,
-    aru: &AvailableRandomizationUnits,
-    app_context: &AppContext,
-) -> Result<()> {
-    log::info!("updating enrollments...");
-    // We might have enrollments for experiments which no longer exist, so we
-    // first build a set of all IDs in both groups.
-    let mut all_slugs = HashSet::new();
-    let experiments = db
-        .get_store(StoreId::Experiments)
-        .collect_all::<Experiment>(writer)?;
+pub(crate) struct EnrollmentsMerger<'a> {
```

Comment:
FWIW, the significance of the name "enrollments merger" is not very clear to me - "merger" suggests that it's going to try to merge and reconcile some state, but what it actually does is evolve from one state to another. It's clunky but "Evolver" might be a better name than "Merger"?